src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -0,0 +1,632 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#include "xdg-autostart-service.h"
+
+#include "conf-parser.h"
+#include "escape.h"
+#include "unit-name.h"
+#include "path-util.h"
+#include "fd-util.h"
+#include "generator.h"
+#include "log.h"
+#include "specifier.h"
+#include "string-util.h"
+#include "nulstr-util.h"
+#include "strv.h"
+
+void xdg_autostart_service_free(XdgAutostartService *s) {
+        if (!s)
+                return;
+
+        free(s->name);
+        free(s->path);
+        free(s->description);
+
+        free(s->type);
+        free(s->exec_string);
+
+        strv_free(s->only_show_in);
+        strv_free(s->not_show_in);
+
+        free(s->try_exec);
+        free(s->autostart_condition);
+        free(s->kde_autostart_condition);
+
+        free(s->gnome_autostart_phase);
+
+        free(s);
+}
```

Comment:
doesn't matter much, but we nowadays prefer to make all destructor functions return NULL cast to the structure they destroy.  this allows you to call it like this:

```c
s = xdg_auto_start_service_free(s);
```

i.e. change the last line of the function to `return mfree(s);` and the function signature to `XdgAutostartService* xdg_autostart_service_free(XdgAutostartService *s)`